### ISSUE TITLE
Rails 5.2.1 seems to be breaking Sandbox migrations

### ISSUE THREAD

**Issue by jacobherrington** [solidusio/solidus#2825]
### Steps to reproduce
From Solidus `bundle exec rake sandbox`

### Expected behavior
A new instance of Sandbox

### Actual behavior
The [remove_order_id_from_inventory_units](https://github.com/solidusio/solidus/blob/425240b84f670558aab61df2655be9a75e00dda0/core/db/migrate/20170319191942_remove_order_id_from_inventory_units.rb#L22) migration fails.

<img width="1008" alt="screen shot 2018-08-08 at 11 22 30 am" src="https://user-images.githubusercontent.com/11466782/43850372-611c71ac-9afd-11e8-8d6f-18b20ef10b6c.png">

Thanks to @VzqzAc for pointing out that the issue originates from https://github.com/activerecord-hackery/polyamorous

It seems we need to set Rails to 5.2.0 to get the migrations working.

### System configuration
**Solidus Version**: 2.7.0.alpha

**Comment by jacobherrington** [comment]
Related: https://github.com/rails/rails/issues/33557

**Comment by VzqzAc** [comment]
Issued on polyamorous/Related?: https://github.com/activerecord-hackery/polyamorous/issues/45

Also seems like any migration with `joins` will fail, not only `remove_order_id_from_inventory_units`

**Comment by pedrofurtado** [comment]
@VzqzAc @jacobherrington The ActiveRecord 5.2.1 changed the number of parameters used by some classes in Private API. Because of this, some gems from activerecord-hackery (ransack, polyamorous, etc.) is broken.

**Comment by kennyadsl** [comment]
Ransack released a new gem version today ([1.8.9](https://rubygems.org/gems/ransack/versions/1.8.9)). Unfortunately they decided (still not sure if that was intentional) to lock Rails compatibility up to 5.1.1. I opened a [new issue on Ransack](https://github.com/activerecord-hackery/ransack/issues/943) to understand better how to go head.

**Comment by kennyadsl** [comment]
To be more clear, this is making impossible for us to support Rails 5.2.1 since ransack 1.8.8 is generating the error above and 1.8.8 does not support Rails 5.2 at all.

**Comment by kennyadsl** [comment]
Working on a PR (#2826).

**Comment by docelic** [comment]
@eduardopatrick to just use Rails 5.2.0 and have it all working with Solidus, there's just a small change you need to do:

1) Create a Rails project with option --skip-bundle or -B added at the end (e.g. "rails new myproject -B", so that it does not run bundler automatically (or if it does run it automatically, just press Ctrl+C to stop it)
2) Then edit file "Gemfile" and find line: "gem 'rails', '~> 5.2.0'" and remove the "~>" so that it becomes "gem 'rails', '5.2.0'
3) Run "bundle" manually
4) And from there you are good to go as you normally would

**Comment by fabioaraujo121** [comment]
Thanks, @docelic ! Saved a big time!

But, anyways, what is really happening with Solidus and Rails 5.2.1?

**Comment by docelic** [comment]
@fabioaraujo121 thanks! From what I see, the PR which we are waiting on which would fix the issue is here: https://github.com/activerecord-hackery/ransack/pull/951

**Comment by louishugens** [comment]
man i can get rid of rails 5.2.1. help please :(

**Comment by docelic** [comment]
@louishugens if you want to remove parts of rails 5.2.1, do as follows:

1) First run `gem list` to see all your installed gems. You might even run `gem list | grep 5.2.1` to narrow the list to gems probably relating to rails 5.2.1
2) For each gem you want to remove, run `gem uninstall GEM_NAME -v=5.2.1`

Hope it helps.

**Comment by tvdeyen** [comment]
The `Gemfile.lock` is a dependency tree.

**Comment by louishugens** [comment]
you are right @tvdeyen, it's rails 5.2.1 that i see in it. how can i get rid of it and replace with rails 5.2.0?

**Comment by docelic** [comment]
According to https://github.com/solidusio/solidus/pull/2826 looks like it works with Rails 5.2.1 now.

Just the source needs to be taken from github and not the release (which isn't out yet), so the instruction from https://github.com/solidusio/solidus to install Solidus by adding the following to `Gemfile`:

```
gem 'solidus'
gem 'solidus_auth_devise'
```

just needs to point to GitHub, i.e. be `gem 'solidus', github: 'solidusio/solidus'`
